pkg/graveler/staging/manager.go

```diff
@@ -4,112 +4,106 @@ import (
 	"context"
 	"errors"
 
-	sq "github.com/Masterminds/squirrel"
-	"github.com/treeverse/lakefs/pkg/db"
 	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/kv"
 	"github.com/treeverse/lakefs/pkg/logging"
 )
 
 type Manager struct {
-	db  db.Database
-	log logging.Logger
+	store kv.StoreMessage
+	log   logging.Logger
 }
 
-func NewManager(db db.Database) *Manager {
-	return &Manager{
-		db:  db,
-		log: logging.Default().WithField("service_name", "postgres_staging_manager"),
+func valueFromProto(pb *graveler.StagedObject) *graveler.Value {
+	return &graveler.Value{
+		Identity: pb.Identity,
+		Data:     pb.Data,
 	}
 }
 
-func (p *Manager) Get(ctx context.Context, st graveler.StagingToken, key graveler.Key) (*graveler.Value, error) {
-	res, err := p.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		value := &graveler.Value{}
-		err := tx.Get(value, "SELECT identity, data FROM graveler_staging_kv WHERE staging_token=$1 AND key=$2", st, key)
-		return value, err
-	}, p.txOpts(db.ReadOnly())...)
-	if errors.Is(err, db.ErrNotFound) {
-		return nil, graveler.ErrNotFound
+func protoFromValue(key []byte, m *graveler.Value) *graveler.StagedObject {
+	return &graveler.StagedObject{
+		Key:      key,
+		Identity: m.Identity,
+		Data:     m.Data,
+	}
+}
+
+func NewManager(store kv.StoreMessage) *Manager {
+	return &Manager{
+		store: store,
+		log:   logging.Default().WithField("service_name", "staging_manager"),
 	}
+}
+
+func (m *Manager) Get(ctx context.Context, st graveler.StagingToken, key graveler.Key) (*graveler.Value, error) {
+	data := &graveler.StagedObject{}
+	_, err := m.store.GetMsg(ctx, string(st), key, data)
 	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrNotFound
+		}
 		return nil, err
 	}
-	value := res.(*graveler.Value)
-	if value.Identity == nil {
+	// Tombstone handling
+	if data.Identity == nil {
 		return nil, nil
 	}
-	return value, nil
+	return valueFromProto(data), nil
 }
 
-func (p *Manager) Set(ctx context.Context, st graveler.StagingToken, key graveler.Key, value *graveler.Value, overwrite bool) error {
+func (m *Manager) Set(ctx context.Context, st graveler.StagingToken, key graveler.Key, value *graveler.Value, overwrite bool) error {
+	// Tombstone handling
```

Comment:
That's not the `WriterFlow` described in the proposal - who's validating that the stagingToken did not change?

Reply:
So we either implement that logic in `graveler` in this PR or open an issue for it.